src/staging/gatsby-node.js

```diff
@@ -254,6 +259,52 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
   await createParkSubPages({ graphql, actions })
   await createSites({ graphql, actions })
   await createPageSlugs("static", staticQuery, { graphql, actions, reporter })
+  await createRedirects({ graphql, actions })
+}
+
+const parkQuery = `
+{
+  allStrapiProtectedArea(filter: {isDisplayed: {eq: true}}) {
+    nodes {
+      id
+      orcs
+      protectedAreaName
+      slug  
+      url
+      oldUrl
+    }
+    totalCount
+  }
+}
+`
+const staticQueryPath = `
+{ 
+  allStrapiLegacyRedirect {
+    nodes {
+      toPath
+      fromPath
+    }
+  }
+}`
+
+async function createRedirects({ graphql, actions, result }) {
+  const response = await strapiApiRequest(graphql, staticQueryPath)
+  const redirects = response.data.allStrapiLegacyRedirect.nodes
+
+  const resultPark = await strapiApiRequest(graphql, parkQuery)
+
+  let parks = resultPark.data.allStrapiProtectedArea.nodes
+
+  parks.map(park => {
+    redirects.map(item => {
+      if (park.slug !== item.toPath) {
+        return actions.createRedirect({
+          fromPath: item.fromPath,
+          toPath: item.toPath,
+        })
+      }
+    })
```

Comment:
This should just be two loops instead of a nested loop

1.  Loop through all the redirects and create a redirect using fromPath and toPath

2. Loop through all the parks and create a redirect for each using "the path from the oldUrl" as the fromPath and the slug as the toPath   (but only if they are different)

e,g, 
- **oldUrl** = https://bcparks.ca/eco_reserve/ambrose_er.html
- **slug** = ecological-reserves/ambrose-lake

therefore
- **fromPath** = /eco_reserve/ambrose_er.html
- **toPath** = ecological-reserves/ambrose-lake